src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -161,83 +165,95 @@ module.exports = function<T, P, I, TI, C>(config : HostConfig<T, P, I, TI, C>) {
   }
 
   function commitAllWork(finishedWork : Fiber) {
-    // Commit all the side-effects within a tree.
-    // First, we'll perform all the host insertions, updates, deletions and
-    // ref unmounts.
-    let effectfulFiber = finishedWork.firstEffect;
-    while (effectfulFiber) {
-      switch (effectfulFiber.effectTag) {
-        case Placement:
-        case PlacementAndCallback: {
-          commitInsertion(effectfulFiber);
-          // Clear the "placement" from effect tag so that we know that this is inserted, before
-          // any life-cycles like componentDidMount gets called.
-          effectfulFiber.effectTag ^= Placement;
-          break;
+    if (!isCommitting) {
+      isCommitting = true;
+      config.prepareForCommit();
+    }
```

Comment:
It realized that this is not actually the correct behavior because if a life-cycle focuses a different thing we don't want to restore it back again.

Since these transaction wrappers are [__before__ the DOM ready queue](
https://github.com/facebook/react/blob/e3131c1d55d6695c2f0966379535f88b813f912b/src/renderers/dom/stack/client/ReactReconcileTransaction.js#L90) and [they execute in order](https://github.com/facebook/react/blob/308e0b77863071f31cff7aec74e359603526274e/src/renderers/shared/stack/reconciler/Transaction.js#L211) it seems to me that the restore should actually happen before the life-cycles get invoked. So this whole block needs to move up to line 214.